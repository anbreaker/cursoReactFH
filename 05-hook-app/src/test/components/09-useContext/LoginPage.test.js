const {mount} = require('enzyme');
const {LoginPage} = require('../../../components/09-useContext/LoginPage');
const {UserContext} = require('../../../components/09-useContext/UserContext');

describe('Pruebas en <LoginPage /> ', () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{setUser}}>
      <LoginPage />
    </UserContext.Provider>
  );

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de ejecutar con el argumento esperado', () => {
    wrapper.find('button').prop('onClick')();
    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      name: 'anbreaker',
    });
  });
});
