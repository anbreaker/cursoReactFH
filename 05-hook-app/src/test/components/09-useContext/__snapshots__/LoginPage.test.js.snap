// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginPage />  debe mostrarse correctamente 1`] = `
<div>
  <div>
    <h1>
      LoginPage
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
